src/RegistryCoordinator.sol

```diff
@@ -498,6 +541,29 @@ contract RegistryCoordinator is EIP712, Initializable, IRegistryCoordinator, ISo
         indexRegistry.deregisterOperator(operatorId, quorumNumbers);
     }
 
+    /**
+     * @notice update operator stake for specified quorumsToUpdate, and deregister if necessary
+     * does nothing if operator is not registered for any quorums.
+     */
+    function _updateOperator(
+        address operator,
+        OperatorInfo memory operatorInfo,
+        bytes memory quorumsToUpdate
+    ) internal {
+        if (operatorInfo.status != OperatorStatus.REGISTERED) {
+            return;
+        }
```

Comment:
I made an earlier comment that we should have this check for both methods and that it didn't cost extra because we're already SLOADing the operatorId

But I'm realizing that's actually not the same slot - the `operatorId` is 32 bytes, so the status is its own SLOAD.

If we want to save an extra SLOAD for each operator updated in `updateOperatorsForQuorum`, we could remove the `REGISTERED` check, put that solely in `updateOperators`, and just pass the `operatorId` into this helper method rather than the entire struct.

Reply:
For weekly avs-sync, 200 operators in quorum0 and 60 operators in 9 remaining quorums, that's ~150,000 gas from the SLOADS. Maybe worth considering removing the check but I also prefer the explicitness